packages/react-components/components/PhoneNumberInput.tsx

```diff
@@ -69,40 +69,56 @@ export default class PhoneNumberInput extends React.Component<Props, State> {
     }
   }
 
+  async getPhoneNumberFromNativePicker() {
+    // linter doesn't allow following code without splitting this code in two lines
+    const number = await SmsRetriever.requestPhoneNumber()
+    return number
+  }
+
   async triggerPhoneNumberRequestAndroid() {
     try {
-      const phone = await SmsRetriever.requestPhoneNumber()
-      const phoneNumber = parsePhoneNumber(phone, '')
+      let phone
+      try {
+        phone = await this.getPhoneNumberFromNativePicker()
+      } catch (error) {
+        console.info(
+          `${TAG}/triggerPhoneNumberRequestAndroid`,
+          'Could not request phone. This might be thrown if the user dismissed the modal',
```

Comment:
Can you differentiate the case when the user dismissed the modal based on the error message thrown by the NativePicker?